src/main/java/com/google/alpollo/AnalysisInfoServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.google.alpollo;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.alpollo.model.AnalysisInfo;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+
+/** Servlet retrieves or saves SongInfo objects to the database. */
+@WebServlet("/analysis-info")
+public class AnalysisInfoServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+
+  /**
+   * Making a GET request to our servlet with the desired song ID as a parameter
+   * will return that song from the database.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long songId = gson.fromJson(request.getReader(), Long.class);
```

Comment:
getReader reads the body of the request, GET requests can't have body, you should get the parameter here as you did before

Reply:
Ok it makes sense now.